AMW_angular/io/src/app/auth/restriction.ts

```diff
@@ -1,8 +1,28 @@
+import { Permission } from './permission';
+
+export enum Action {
```

Comment:
Did you consider using string unions? For example:
export type Action = "READ" | "CREATE" ...`?
This would guarantee the same type safety while being more lightweight.